pkg/configuration/configuration.go

```diff
@@ -33,10 +43,182 @@ const (
+	cfg := GetCachedRegistrationServiceConfig()
+	return cfg.Environment() == UnitTestsEnvironment
 }
 
 func Namespace() string {
 	return os.Getenv(commonconfig.WatchNamespaceEnvVar)
 }
+
+// GetRegistrationServiceConfig returns a RegistrationServiceConfig using the cache, or if the cache was not initialized
+// then retrieves the latest config using the provided client and updates the cache
+func GetRegistrationServiceConfig(cl client.Client) (RegistrationServiceConfig, error) {
+	config, secrets, err := commonconfig.GetConfig(cl, &toolchainv1alpha1.ToolchainConfig{})
+	if err != nil {
+		// return default config
+		logger.Error(err, "failed to retrieve RegistrationServiceConfig")
+		return RegistrationServiceConfig{cfg: &toolchainv1alpha1.ToolchainConfigSpec{}}, err
+	}
+	return NewRegistrationServiceConfig(config, secrets), nil
+}
+
+// GetCachedRegistrationServiceConfig returns a RegistrationServiceConfig directly from the cache
+func GetCachedRegistrationServiceConfig() RegistrationServiceConfig {
+	config, secrets := commonconfig.GetCachedConfig()
+	return NewRegistrationServiceConfig(config, secrets)
+}
+
+// ForceLoadRegistrationServiceConfig updates the cache using the provided client and returns the latest RegistrationServiceConfig
+func ForceLoadRegistrationServiceConfig(cl client.Client) (RegistrationServiceConfig, error) {
+	config, secrets, err := commonconfig.LoadLatest(cl, &toolchainv1alpha1.ToolchainConfig{})
+	if err != nil {
+		// return default config
+		logger.Error(err, "failed to force load RegistrationServiceConfig")
+		return RegistrationServiceConfig{cfg: &toolchainv1alpha1.ToolchainConfigSpec{}}, err
+	}
+	return NewRegistrationServiceConfig(config, secrets), nil
+}
+
+type RegistrationServiceConfig struct {
+	cfg     *toolchainv1alpha1.ToolchainConfigSpec
+	secrets map[string]map[string]string
+}
+
+func NewRegistrationServiceConfig(config runtime.Object, secrets map[string]map[string]string) RegistrationServiceConfig {
+	if config == nil {
+		// return default config if there's no config resource
+		return RegistrationServiceConfig{cfg: &toolchainv1alpha1.ToolchainConfigSpec{}}
+	}
+
+	toolchaincfg, ok := config.(*toolchainv1alpha1.ToolchainConfig)
+	if !ok {
+		// return default config
+		logger.Error(fmt.Errorf("cache does not contain toolchainconfig resource type"), "failed to get ToolchainConfig from resource, using default configuration")
+		return RegistrationServiceConfig{cfg: &toolchainv1alpha1.ToolchainConfigSpec{}}
+	}
+	return RegistrationServiceConfig{cfg: &toolchaincfg.Spec, secrets: secrets}
+}
+
+func (r *RegistrationServiceConfig) Print() {
+	if r.cfg == nil {
+		logger.Info("ToolchainConfig not found, using default Registration Service configuration")
+		return
+	}
+	logger.Info("Registration Service Configuration")
+	indentedJSON, _ := json.MarshalIndent(r.cfg.Host.RegistrationService, "", "\t")
+	fmt.Println(string(indentedJSON))
```

Comment:
Shouldn't be the config printed out using the logger? Something like that
```suggestion
	indentedJSON, _ := json.MarshalIndent(r.cfg.Host.RegistrationService, "", "\t")
	logger.Info("Registration Service Configuration", "config", string(indentedJSON))
```
I didn't test it though.